mrblib/opt_parser.rb

```diff
@@ -40,19 +40,23 @@ def initialize
 
   # Add a flag and a callback to invoke if flag is given later.
   #
-  # @param [ String ] flag The name of the option value.
-  #                        Possible values: object, string, int, float, bool
+  # @param [ String ] opt The name of the option value.
   # @param [ Symbol ] type The type of the option v
-  # @param [ Object ] dval The value to use if nothing else given.
+  #                        Possible values: object, string, int, float, bool
+  # @param [ Object ] default The value to use if nothing else given.
+  # @param [ String ] short The short name of the option value
   # @param [ Proc ]   blk  The callback to be invoked.
   #
   # @return [ Void ]
-  def on(opt, type = :object, dval = nil, &blk)
+  def on(opt, type = :object, default: nil, short: nil, &blk)
```

Comment:
I dont like the breaking change to be honest. The syntax is propably better and more flexible but not backward compatible.

I think thats better:

```ruby
def on(opt, type = :object, dval = nil, short: nil, &blk)
```